x-pack/plugins/cases/server/client/user_actions/users.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { GetCaseUsersResponse } from '../../../common/api';
+import { GetCaseUsersResponseRt } from '../../../common/api';
+import type { OwnerEntity } from '../../authorization';
+import { Operations } from '../../authorization';
+import { createCaseError } from '../../common/error';
+import type { CasesClient } from '../client';
+import type { CasesClientArgs } from '../types';
+import type { GetUsersRequest } from './types';
+
+export const getUsers = async (
+  { caseId }: GetUsersRequest,
+  casesClient: CasesClient,
+  clientArgs: CasesClientArgs
+): Promise<GetCaseUsersResponse> => {
+  const {
+    services: { userActionService },
+    logger,
+    authorization,
+  } = clientArgs;
+
+  try {
+    const entities: OwnerEntity[] = [];
+    const participants: GetCaseUsersResponse['participants'] = [];
+
+    // ensure that we have authorization for reading the case
+    await casesClient.cases.resolve({ id: caseId, includeComments: false });
```

Comment:
In the description of this PR you mention that this API will not return the current assignees and instead we'll rely on the UI to combine the assignees with the results of this API.

Do you foresee us needing to retrieving all users who have been associated with a case? Meaning all users who have performed an action and ever been assigned? If so, I think it might better to create two apis, one that returns all participants and one that returns all users.

Since we're getting the case here anyway for authorization why not grab the assignees from here and return them as part of the response?

Since we defined a participant as anyone who has performed an action on the case and the current assignees, I think this would allow us to avoid the terms aggregation on the assignees in the service layer:

```
field: `${CASE_USER_ACTION_SAVED_OBJECT}.attributes.payload.assignees.uid`
```

Instead we can use the assignee results from the resolve call.

I feel like this simplify the consumption of this API because the UI can just rely on this API for the source of truth for the participants.

If we go with this model I think the response could look like:

```
{
    "participants": [
        {
            "username": "cases_all",
            "full_name": "",
            "email": "",
            "profile_uid": "u_4okFGH1aToaMxNh2ZWzCQkwoQUiPEuMBE_RdnXd8vC4_0"
        },
        {
            "username": "elastic",
            "full_name": null,
            "email": null,
            "profile_uid": "u_mGBROF_q5bmFCATbLXAcCwKa0k8JvONAwSruelyKA5E_0"
        },
        {
            "profile_uid": "u_3OgKOf-ogtr8kJ5B0fnRcqzXs2aQQkZLtzKEEFnKaYg_0"
        }
    ]
}
```

If we go that route we can also name the API `/_participants`.

Reply:
What you suggest makes total sense and I would love to format the response as you suggested. The only reason I separate participants from users (and did a terms aggregation on the assignees) is that if you assign a user to a case and de-assign him/her then the user is not part of the assignees (case SO). The UI needs to know the profile UID of the user that got de-assign so it can display the user's information in the "removed assignee" user action.